sycl/include/sycl/detail/pi.h

```diff
@@ -264,6 +264,7 @@ typedef enum {
   // Intel UUID extension.
   PI_DEVICE_INFO_UUID = 0x106A,
   // These are Intel-specific extensions.
+  PI_DEVICE_INFO_DEVICE_ID = 0x4251,
```

Comment:
The value here is the same as the value of `CL_DEVICE_ID_INTEL` which is defined in the OpenCL extension `cl_intel_device_attribute_query`.  I presume the intention is that the OpenCL plugin doesn't need any changes because we can pass `PI_DEVICE_INFO_DEVICE_ID` in place of `CL_DEVICE_ID_INTEL` and it will just work.

However, I don't see any code checking to see if the OpenCL implementation supports the `cl_intel_device_attribute_query` extension.  Is that consistent with the way we use other OpenCL extensions?  Do we just assume that OpenCL supports whatever extension we need without checking?

Reply:
[clGetDeviceInfo](https://registry.khronos.org/OpenCL/sdk/2.2/docs/man/html/clGetDeviceInfo.html) states that it returns

> CL_​INVALID_​VALUE if param_name is not one of the supported values or if size in bytes specified by param_value_size is < size of return type as specified in the [Device Queries](https://www.khronos.org/registry/OpenCL/specs/2.2/html/OpenCL_API.html#device-queries-table) table and param_value is not a NULL value or if *param_name is a value that is available as an extension and the corresponding extension is not supported by the device.*

The runtime uses this to determine if the aspect for each of the extension descriptors is supported or not. So OpenCL should be ready out-of-the-gate.

Reply:
> I presume the intention is that the OpenCL plugin doesn't need any changes because we can pass PI_DEVICE_INFO_DEVICE_ID in place of CL_DEVICE_ID_INTEL and it will just work.

Yes, that is the intention.

Reply:
Could you please be a bit more specific about what kind of documentation do you expect here? Most other constants in `pi.h` also map to an OpenCL parameter the same way, some of them are part of the core OpenCL but many others map to extensions. Neither of those are documented. So there's nothing new in the way `PI_DEVICE_INFO_DEVICE_ID` works here. I followed the conventions of the existing codebase.

Reply:
Ah, OK. I was under a wrong impression that we are only mapping PI to CL values in the OpenCL plugin, and this new one had a direct CL value used, which I thought needs a documentation. Nevermind.